### ISSUE TITLE
mantle Register not working

### ISSUE THREAD

**Issue by rsetaluri** [phanrahan/mantle#107]
on latest master of both magma and mantle.

```
>>> import magma
>>> import mantle
>>> 
>>> mantle.Register(8, has_ce=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/setaluri/dev/mantle/mantle/coreir/register.py", line 37, in Register
    has_async_reset)(**kwargs)
  File "/Users/setaluri/dev/mantle/mantle/coreir/register.py", line 14, in DefineRegister
    class Register(m.Circuit):
  File "/Users/setaluri/dev/magma/magma/circuit.py", line 455, in __new__
    self.definition()
  File "/Users/setaluri/dev/mantle/mantle/coreir/register.py", line 26, in definition
    I = Mux(n)(reg.O, I, circ.CE)
NameError: name 'Mux' is not defined
```

**Comment by rsetaluri** [comment]
Actually seems to be a lot of issues in this file. Will push a fix.

**Comment by leonardt** [comment]
Fixed by #108, thanks!
